cmake/modules/FindBOOST.cmake

```diff
@@ -78,14 +78,54 @@ IF(Boost_FOUND)
   SET(BOOST_VERSION
     "${BOOST_VERSION_MAJOR}.${BOOST_VERSION_MINOR}.${BOOST_VERSION_SUBMINOR}"
     )
+
+  #
+  # Test that Boost.Iostreams is usuable.
+  #
+  RESET_CMAKE_REQUIRED()
+  PUSH_CMAKE_REQUIRED("${DEAL_II_CXX_VERSION_FLAG}")
+  PUSH_CMAKE_REQUIRED("-L${Boost_LIBRARY_DIRS}")
+  SET(CMAKE_REQUIRED_LIBRARIES "-lboost_iostreams")
+  CHECK_CXX_SOURCE_COMPILES(
+    "
+    #include <string>
+    #include <boost/iostreams/device/back_inserter.hpp>
+    #include <boost/iostreams/filter/gzip.hpp>
+    #include <boost/iostreams/filtering_stream.hpp>
+
+    int main()
+    {
+      std::string decompressed_buffer;
+      char test[1] = {'c'};
+
+      boost::iostreams::filtering_ostream decompressing_stream;
+      decompressing_stream.push(boost::iostreams::gzip_decompressor());
+      decompressing_stream.push(boost::iostreams::back_inserter(decompressed_buffer));
+      decompressing_stream.write (test, 1);
+    }
+    "
+    DEAL_II_BOOST_IOSTREAMS_USUABLE
+    )
+  MESSAGE(STATUS "${DEAL_II_BOOST_IOSTREAMS_USUABLE}")
+  IF(${DEAL_II_BOOST_IOSTREAMS_USUABLE})
+    MESSAGE(STATUS "Boost.Iostreams is usuable.")
+  ELSE()
+      MESSAGE(STATUS "Boost.Iostreams is not usuable. "
```

Comment:
I would say we need to throw an error if user provided `-DBOOST_DIR` at configure and we now are falling back to internal boost disregarding this input.

But I don't feel strong about it.

Reply:
Me neither. I can change this if someone feels strong about it.